tests/test_sensor.py

```diff
@@ -262,3 +262,55 @@ def test_remove_callbacks(self, ss, sensors, add_callback, remove_callback):
         ss.remove(sensors[0])
         remove_callback.assert_not_called()
         add_callback.assert_not_called()
+
+
+class MyAgg(AggregateSensor):
+    def update_aggregate(self, updated_sensor, reading, old_reading):
+        return Reading(0, Sensor.Status.NOMINAL, 7)
+
+
+def test_aggregate_sensor(mocker, ss, sensors):
+    """Test operation of AggregateSensor.
+
+    From a simple derived class, mock out the _update_aggregate function, and
+    test that (1) the target is properly assigned, and (2) the update function
+    is called at all appropriate occasions.
+    """
+
+    # Check that creation happens properly, and correct initial value is set.
+    my_agg = MyAgg(target=ss, sensor_type=int, name="good-bad-ugly")
+    assert my_agg.target is ss
+    assert my_agg.reading.timestamp == 0
+    assert my_agg.reading.status == Sensor.Status.NOMINAL
+    assert my_agg.reading.value == 7
+
+    # Mock out update_aggregate so we can check it's called appropriately.
+    def side_effect(*args, **kwargs):
+        """Return an arbitrary Reading."""
+        return Reading(0, Sensor.Status.WARN, 8)
+
+    mocker.patch.object(my_agg, "update_aggregate", side_effect=side_effect)
+
+    # Check that update function is called when a sensor is added.
+    ss.add(sensors[1])
+    my_agg.update_aggregate.assert_called_with(sensors[1], sensors[1].reading, None)
+    assert my_agg.reading.timestamp == 0
+    assert my_agg.reading.status == Sensor.Status.WARN
+    assert my_agg.reading.value == 8
+
+    # Check that it's called for a removed sensor,
+    ss.remove(sensors[0])
+    my_agg.update_aggregate.assert_called_with(sensors[0], None, sensors[0].reading)
```

Comment:
Maybe also check that the update doesn't get called if the value changes after the sensor is removed.